crates/bevy_reflect/src/array.rs

```diff
@@ -191,26 +188,25 @@ impl DynamicArray {
         }
     }
 
-    #[inline]
-    pub fn name(&self) -> &str {
-        &self.name
-    }
-
-    #[inline]
-    pub fn set_name(&mut self, name: String) {
-        self.name = name;
+    /// Sets the [type] to be represented by this `DynamicArray`.
+    ///
+    /// [type]: TypeInfo
+    pub fn set_represented_type(&mut self, represented_type: Option<&'static TypeInfo>) {
+        self.represented_type = represented_type;
```

Comment:
should there be an assertion here that `represented_type` is a `TypeInfo::Array`?

Reply:
Hm good idea. Otherwise, things can go very wrong very fast haha.

I'll update the other dynamic types in the same way

Reply:
we might also want to verify the types of fields of dynamic types in a similar way, but this might have performance concerns.

it might also make sense (in a future PR) to recursively try and set the represented types of dynamic fields, but that would require being able to call `set_represented_type` from just a dyn Reflect` and i don't think downcasting sufficient.

one possible way of doing this might be to replace `set_represented_type` with `get_represented_type_mut` and return an enum:
```rust
enum RepresentedTypeMut<'a> {
    Dynamic(&'a mut Option<&'static TypeInfo>),
    Concrete(&'static TypeInfo),
}
```

Reply:
Yeah the recursive check could be useful, but I think it does add to the overall cost of things like cloning, which needs to recurse into the fields as well.

We can definitely come back to this in a future PR, though!